src/lib.rs

```diff
@@ -209,13 +216,23 @@ pub async fn start_lemmy_server(args: CmdArgs) -> LemmyResult<()> {
   } else {
     None
   };
+
+  // This FederationConfig instance is exclusively used to send activities, so we can safely
+  // increase the timeout without affecting timeouts for resolving objects anywhere.
+  let federation_sender_config = if !args.disable_activity_sending {
```

Comment:
I've tried to embed this in the `then()` block first, but that is not an async block and turning it into async seems to be a bit more complex as it expects a `CancellableTask` there.